contracts/MechMarketplace.sol

```diff
@@ -177,30 +175,33 @@ contract MechMarketplace is IErrorsMarketplace {
         );
     }
 
-    /// @dev Calculates payment for request delivery.
+    /// @dev Calculates payment and fee based on delivery rates and mech type.
     /// @param mech Delivery mech address.
-    /// @param payment Payment amount.
+    /// @param deliveryRate Delivery rate.
+    /// @param mechPayment Mech payment.
+    /// @param marketplaceFee Marketplace fee.
     function _calculatePayment(
```

Comment:
This becomes a single calculation based on deliveryRate. Clearly can be adjusted further in its final escrow function as it is virtual.